src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala

```diff
@@ -0,0 +1,195 @@
+    getStructType(xmlSchema)
+  }
+
+  /**
+   * Reads a schema from an XSD file.
+   *
+   * @param xsdFile XSD file
+   * @return Spark-compatible schema
+   */
+  @Experimental
+  def read(xsdFile: Path): StructType = read(xsdFile.toFile)
+
+  /**
+   * Reads a schema from an XSD as a string.
+   *
+   * @param xsdString XSD as a string
+   * @return Spark-compatible schema
+   */
+  @Experimental
+  def read(xsdString: String): StructType = {
+    val xmlSchema = new XmlSchemaCollection().read(new StringReader(xsdString))
+    getStructType(xmlSchema)
+  }
+
+
+  private def getStructField(xmlSchema: XmlSchema, schemaType: XmlSchemaType): StructField = {
+    schemaType match {
+      // xs:simpleType
+      case simpleType: XmlSchemaSimpleType =>
+        val baseName = "baseName"
+        simpleType.getContent match {
+          case restriction: XmlSchemaSimpleTypeRestriction =>
+            val matchType =
+              if (restriction.getBaseTypeName == Constants.XSD_ANYSIMPLETYPE) {
+                simpleType.getQName
+              } else {
+                restriction.getBaseTypeName
+              }
+            matchType match {
+              case Constants.XSD_BOOLEAN => StructField(baseName, BooleanType)
+              case Constants.XSD_BYTE => StructField(baseName, BinaryType)
+              case Constants.XSD_DATE |
+                   Constants.XSD_DATETIME |
+                   Constants.XSD_TIME =>
+                StructField(baseName, StringType)
+              case Constants.XSD_DECIMAL =>
+                val scale = restriction.getFacets.asScala.collectFirst {
+                  case facet: XmlSchemaFractionDigitsFacet => facet
+                }
+                scale match {
+                  case Some(scale) => StructField(
+                    baseName, DecimalType(38, scale.getValue.toString.toInt))
+                  case None => StructField(baseName, DecimalType(38, 18))
+                }
+              case Constants.XSD_DOUBLE => StructField(baseName, DoubleType)
+              case Constants.XSD_FLOAT => StructField(baseName, FloatType)
+              case Constants.XSD_INTEGER |
+                   Constants.XSD_SHORT |
+                   Constants.XSD_NEGATIVEINTEGER |
+                   Constants.XSD_NONNEGATIVEINTEGER |
+                   Constants.XSD_NONPOSITIVEINTEGER |
+                   Constants.XSD_POSITIVEINTEGER |
+                   Constants.XSD_UNSIGNEDINT |
+                   Constants.XSD_UNSIGNEDLONG |
+                   Constants.XSD_UNSIGNEDSHORT =>
+                StructField(baseName, IntegerType)
+              case Constants.XSD_LONG => StructField(baseName, LongType)
+              case Constants.XSD_STRING |
+                   Constants.XSD_BASE64 |
+                   Constants.XSD_ANYTYPE =>
+                StructField(baseName, StringType)
+            }
+          case _ => StructField(baseName, StringType)
```

Comment:
This is about the only substantive change, which helps with 'any' type fields with no further content restrictions. Just treat them as strings. Other changes to the code are cosmetic simplifications (IMHO)

Reply:
Agree.